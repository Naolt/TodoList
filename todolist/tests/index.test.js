import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import Home from "../pages/index";

describe("Home component", () => {
  beforeEach(() => {
    // Clear localStorage before each test
    localStorage.clear();
  });

  test("renders the Todo List component", () => {
    render(<Home />);

    // Check if the Todo List heading is rendered
    const headingElement = screen.getByText(/Todo List/i);
    expect(headingElement).toBeInTheDocument();

    // Check if the Add Task input is rendered
    const addTaskInput = screen.getByTestId("add-task-input");
    expect(addTaskInput).toBeInTheDocument();

    // Check if the Add Task button is rendered
    const addTaskButton = screen.getByTestId("add-task-button");
    expect(addTaskButton).toBeInTheDocument();
  });

  test("adds a new task to the list", () => {
    render(<Home />);

    // Add a new task
    const addTaskInput = screen.getByTestId("add-task-input");
    const addTaskButton = screen.getByTestId("add-task-button");
    fireEvent.change(addTaskInput, { target: { value: "New Task" } });
    fireEvent.click(addTaskButton);

    // Check if the new task is added to the list
    const taskTitle = screen.getByTestId("task-title-1");
    expect(taskTitle).toBeInTheDocument();
  });

  test("toggles task completion when checkbox is clicked", () => {
    render(<Home />);

    // Add a new task
    const addTaskInput = screen.getByTestId("add-task-input");
    const addTaskButton = screen.getByTestId("add-task-button");
    fireEvent.change(addTaskInput, { target: { value: "New Task" } });
    fireEvent.click(addTaskButton);

    // Click the checkbox to toggle completion
    const checkbox = screen.getByTestId("checkbox-task-1");
    fireEvent.click(checkbox);

    // Check if the task is marked as completed
    expect(checkbox).toHaveProperty("checked", true);
  });

  test("deletes a task when delete button is clicked", () => {
    render(<Home />);

    // Add a new task
    const addTaskInput = screen.getByTestId("add-task-input");
    const addTaskButton = screen.getByTestId("add-task-button");
    fireEvent.change(addTaskInput, { target: { value: "New Task" } });
    fireEvent.click(addTaskButton);

    // Click the delete button to delete the task
    const deleteButton = screen.getByTestId("delete-task-1");
    fireEvent.click(deleteButton);

    // Check if the task is deleted from the list
    const taskTitle = screen.queryByText("New Task");
    expect(taskTitle).not.toBeInTheDocument();
  });
});
