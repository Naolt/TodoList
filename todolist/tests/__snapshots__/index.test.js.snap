// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders Home component correctly 1`] = `
<div>
  <div
    style="text-align: center; background-color: rgb(245, 245, 245); padding: 2rem; height: 100vh; overflow: hidden;"
  >
    <h1
      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1o2xzv0-MuiTypography-root"
    >
      Todo List
    </h1>
    <div
      class="MuiBox-root css-1embff6"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-o9dsj-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r7:"
          id=":r7:-label"
        >
          Add Task
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
            data-testid="add-task-input"
            id=":r7:"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Add Task
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-144pftz-MuiButtonBase-root-MuiButton-root"
        data-testid="add-task-button"
        disabled=""
        tabindex="-1"
        type="button"
      >
        Add
      </button>
    </div>
    <ul
      class="MuiList-root MuiList-padding MuiList-dense css-12wg1bs-MuiList-root"
    >
      <div />
    </ul>
  </div>
</div>
`;
